bpf/lib/nat.h

```diff
@@ -338,9 +339,10 @@ static __always_inline int snat_v4_track_connection(struct __ctx_buff *ctx,
 	memcpy(&tmp, tuple, sizeof(tmp));
 
 	where = dir == NAT_DIR_INGRESS ? CT_INGRESS : CT_EGRESS;
+	ipv4_ct_tuple_swap_ports(&tmp);
```

Comment:
Mind adding a short comment here why the swap is needed? (the callers extract the ports as-is, but CT wants tuples with swapped tuples)

Reply:
My thinking would be that it's good to have the calling path explicitly say "yes, this case applies to me". Just because all this stuff has proven to be super-tricky, and we're definitely going to get it wrong again in the future... :).